runtime/lua/vim/_init_packages.lua

```diff
@@ -40,6 +40,21 @@ function vim._load_package(name)
     local f, err = package.loadlib(found[1], 'luaopen_' .. modname:gsub('%.', '_'))
     return f or error(err)
   end
+
+  if vim.startswith(basename, 'vim/') then
+    local vimruntime = os.getenv('VIMRUNTIME')
+    if vimruntime then
+      local accessible = vim.fn.isdirectory(vimruntime) == 1
+        and vim.fn.filereadable(vim.fn.expand(vimruntime)) == 1
+      if not accessible then
+        error('$VIMRUNTIME is not accessible or has insufficient permissions')
+      else
+        error('Module or shared library not found for ' .. name)
```

Comment:
when we don't know any specific error, we should return `nil` (the package loader will let the search continue and display the final error mesage)

Reply:
if we want validate it in main function i think loader no need change then ..